### PR TITLE
*: Support "show grants for current_user();"

### PR DESCRIPTION
**Pull request by shenli**
Fix issue https://github.com/pingcap/tidb/issues/5691

### CODE REVIEW

**Comment by winkyao** [comment]
Plz fix ci.

**Comment by shenli** [comment]
I will fix this.

**Comment by zimulala** [comment]
@shenli 
Any update?

**Comment by shenli** [comment]
Blocked by https://github.com/pingcap/tidb/pull/6154

**Comment by shenli** [comment]
@zimulala @winkyao @tiancaiamao CI is fixed.

File: executor/show_test.go [hunk 1]
```diff
@@ -549,6 +549,9 @@ func (s *testSuite) TestShow2(c *C) {
 
 	tk.MustExec(`grant all on *.* to 'root'@'%'`)
 	tk.MustQuery("show grants").Check(testkit.Rows(`GRANT ALL PRIVILEGES ON *.* TO 'root'@'%'`))
+
+	tk.MustQuery("show grants for current_user()").Check(testkit.Rows("GRANT ALL PRIVILEGES ON *.* TO 'root'@'%'"))
```
Comment (ngaut) [inline executor/show_test.go]: Please use back quote instead of double quote.

File: executor/show_test.go [hunk 2]
```diff
@@ -549,6 +549,9 @@ func (s *testSuite) TestShow2(c *C) {
...
+	tk.MustQuery("show grants for current_user").Check(testkit.Rows("GRANT ALL PRIVILEGES ON *.* TO 'root'@'%'"))
```
Comment (ngaut) [inline executor/show_test.go]: Ditto.

**Comment by shenli** [comment]
@ngaut PTAL

**Comment by ngaut** [comment]
LGTM

File: plan/preprocess.go [hunk 3]
```diff
@@ -423,7 +423,7 @@ func checkColumn(colDef *ast.ColumnDef) error {
 	return nil
 }
 
-// isNowSymFunc checks whether default value is a NOW() builtin function.
+// isDefaultValNowSymFunc checks whether defaul value is a NOW() builtin function.
```
Comment (tiancaiamao) [inline plan/preprocess.go]: s/defaul/default

**Pull request merged**